api/api.go

```diff
@@ -22,12 +25,26 @@ func Setup(ctx context.Context, r *mux.Router, permissions AuthHandler) *API {
 		Router: r,
 	}
 
-	nopSession := &NOPSession{}
-	nopCache := &NOPCache{}
+	s := session.NewSession()
 
-	r.HandleFunc("/sessions", permissions.Require(create, CreateSessionHandlerFunc(nopSession, nopCache))).Methods("POST")
-	r.HandleFunc("/sessions/{ID}", GetByIDSessionHandlerFunc(nopCache, mux.Vars)).Methods("GET")
-	r.HandleFunc("/sessions", permissions.Require(delete, DeleteAllSessionsHandlerFunc(nopCache))).Methods("DELETE")
+	cfg, err := config.Get()
```

Comment:
Setting up session and then cache should be in service.Runc() function

Tip: Never call `config.Get()` twice in an application, if someone updates the configuration different parts of the app will have used different configurations, even worse if these are then stored as new parameters. This can lead to querky behaviour at best and data issues at worst or app failure.

Reply:
I would move lines 30 to 44 into `service.Run()` func in `service/service.go`, then pass the session and cache to the api.